### PR TITLE
Docs: update Installation and Auto-provisioning GPUs.

### PR DESCRIPTION
**Pull request by concretevitamin**
Installation
- Make installation commands stand out 
- Highlight Nightly over Latest Release, using tabs
- Move `sky check` section up to immediately follow install
- Update docker image to the debug image, which clones the main branch

Auto-provisioning GPUs
- Basics of provisioning GPUs

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below): rendered locally
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: docs/source/getting-started/installation.rst [hunk 1]
```diff
@@ -7,61 +7,122 @@ Installation
 
     For Macs, macOS >= 10.15 is required to install SkyPilot. Apple Silicon-based devices (e.g. Apple M1) must run :code:`pip uninstall grpcio; conda install -c conda-forge grpcio=1.43.0` prior to installing SkyPilot.
 
-
 Install SkyPilot using pip:
 
-.. code-block:: console
+.. tab-set::
```
Comment (Michaelvll) [inline docs/source/getting-started/installation.rst:12]: Nice!

File: docs/source/getting-started/installation.rst [hunk 2]
```diff
@@ -7,61 +7,122 @@ Installation
...
-  $ pip install -U "skypilot-nightly[all]"
+.. code-block:: shell
 
-or install from source:
+  # Nightly:
+  pip install -U "skypilot-nightly[aws,gcp]"
+  # Latest release:
+  pip install -U "skypilot[aws,gcp]"
+
+To get the latest features, either use a nightly build (the **Nightly**
```
Comment (Michaelvll) [inline docs/source/getting-started/installation.rst:68]: Nit: Wondering should we have the source in the tabs above as well?

File: docs/source/getting-started/installation.rst [hunk 3]
```diff
@@ -7,61 +7,122 @@ Installation
...
 
-.. note::
+Verifying cloud access
+------------------------------------
+
+After installation, run :code:`sky check` to verify that credentials are correctly set up:
+
+.. code-block:: console
+
+  $ sky check
```
Comment (Michaelvll) [inline docs/source/getting-started/installation.rst:88]: We are having `$` for all the commands below, but not for the command for installation above. Is that expected?

File: docs/source/getting-started/installation.rst [hunk 4]
```diff
@@ -300,28 +335,38 @@ increases before proceeding.
...
 .. code-block:: console
 
-  $ docker run -td --name sky --rm -v "$HOME/.sky:/root/.sky:rw" -v "$HOME/.aws:/root/.aws:rw" -v "$HOME/.config/gcloud:/root/.config/gcloud:rw" berkeleyskypilot/skypilot:latest
+  $ # NOTE: '--platform linux/amd64' is needed for Apple silicon Macs
+  $ docker run --platform linux/amd64 \
+    -td --rm --name sky -e SKYPILOT_DEBUG=0 \
+    -v "$HOME/.sky:/root/.sky:rw" \
+    -v "$HOME/.aws:/root/.aws:rw" \
+    -v "$HOME/.config/gcloud:/root/.config/gcloud:rw" \
+    berkeleyskypilot/skypilot-debug
```
Comment (Michaelvll) [inline docs/source/getting-started/installation.rst:350]: Should we update this to `skypilot-nightly`?

File: docs/source/getting-started/installation.rst [hunk 5]
```diff
@@ -300,28 +335,38 @@ increases before proceeding.
...
 
-If your cloud CLIs are already setup, your credentials will be mounted to the
-container and you can proceed to :ref:`Quickstart <quickstart>`. Else you can
-follow the instructions in :ref:`Cloud account setup <cloud-account-setup>`
-inside the container to setup your cloud accounts.
+  $ # Run inside the container:
 
-If you exit from the shell in the container, the container will keep running
-in the background. You can start a new shell with :code:`docker exec -it sky /bin/bash`.
+  $ cd /sky_repo/skypilot
```
Comment (Michaelvll) [inline docs/source/getting-started/installation.rst:355]: These lines are not needed with the nightly docker image

File: docs/source/getting-started/installation.rst [hunk 6]
```diff
@@ -332,6 +377,10 @@ Finally, you can stop the container with:
 
   $ docker stop sky
 
+See more details about the dev container image
```
Comment (Michaelvll) [inline docs/source/getting-started/installation.rst:380]: This needs to be updated?

**Review by Michaelvll — COMMENTED** [review]
Thanks for updating the doc @concretevitamin ! The pr mostly looks good to me. Left several nits.

File: docs/source/getting-started/installation.rst (hunk 2 above)
Comment (concretevitamin) [reply docs/source/getting-started/installation.rst:79]: Good call, done.

File: docs/source/getting-started/installation.rst (hunk 3 above)
Comment (concretevitamin) [reply docs/source/getting-started/installation.rst]: Changed console -> shell and removed `$` everywhere on this page. Feels cleaner.

File: docs/source/getting-started/installation.rst (hunk 4 above)
Comment (concretevitamin) [reply docs/source/getting-started/installation.rst]: Done.

File: docs/source/getting-started/installation.rst (hunk 5 above)
Comment (concretevitamin) [reply docs/source/getting-started/installation.rst]: Done.

File: docs/source/getting-started/installation.rst (hunk 6 above)
Comment (concretevitamin) [reply docs/source/getting-started/installation.rst:386]: Done.

File: docs/source/getting-started/installation.rst (hunk 2 above)
Comment (Michaelvll) [reply docs/source/getting-started/installation.rst:79]: Oh, I was wondering if "install from source" can be another tab as `From Source` in the installation instruction above?

**Review by Michaelvll — APPROVED** [review]
Thanks for the update @concretevitamin! LGTM with 2 nits.

File: docs/source/getting-started/installation.rst (hunk 2 above)
Comment (concretevitamin) [reply docs/source/getting-started/installation.rst]: Done.

File: docs/source/getting-started/installation.rst (hunk 3 above)
Comment (concretevitamin) [reply docs/source/getting-started/installation.rst]: Good point. The install page needs to reduce friction/text as much as possible, so opting to not keep `$`. For other pages, there's already some inconsistency which I think is okay (if some snippets have both commands and outputs, then it makes sense to `$`).

**Pull request merged**